plugin/al/lib/AL_USDMaya/AL/usdmaya/fileio/translators/TranslatorBase.h

```diff
@@ -460,11 +466,30 @@ class TranslatorManufacture
     /// \return returns the python translator (if one is available)
     static TranslatorRefPtr getPythonTranslatorBySchemaType(const TfToken type_name);
 
+    /// \brief set contexts of all python translators.
+    static void setPythonTranslatorContexts(TranslatorContext::RefPtr context);
+
     std::unordered_map<std::string, TranslatorRefPtr>        m_translatorsMap;
     static std::unordered_map<std::string, TranslatorRefPtr> m_assetTypeToPythonTranslatorsMap;
     std::vector<ExtraDataPluginPtr>                          m_extraDataPlugins;
     static TranslatorRefPtrVector                            m_pythonTranslators;
     TranslatorRefPtrVector                                   m_contextualisedPythonTranslators;
+
+    static TranslatorContextPtrStack m_contextPtrStack;
+};
+
+//----------------------------------------------------------------------------------------------------------------------
+/// \brief  used to prepare python translator context on creation and push it into stack, and pop
+/// from stack on destruction. \ingroup   translators
+//----------------------------------------------------------------------------------------------------------------------
+class TranslatorContextSetterCtx
+{
+public:
+    TranslatorContextSetterCtx(TranslatorContext::RefPtr context)
+    {
+        TranslatorManufacture::preparePythonTranslators(context);
```

Comment:
I'm assuming this code is designed to run single-threaded? I'm wondering if we have somewhere code available to assert that we execute on main-thread...

Reply:
Yes, I think so. We have used this code for a long time internally but no threading issue / race condition problem so far.